### PR TITLE
APPS-1047 Add time filters to the list of full backups endpoint

### PR DESCRIPTION
**Pull request by reugn**

### CODE REVIEW

File: pkg/service/backup_backend_local.go [hunk 1]
```diff
@@ -111,7 +111,9 @@ func (local *BackupBackendLocal) FullBackupList() ([]model.BackupDetails, error)
 	for _, e := range entries {
 		if e.IsDir() {
 			details := toBackupDetails(e, backupFolder)
-			if details.LastModified.Before(lastRun) {
+			if details.LastModified.Before(lastRun) &&
```
Comment (korotkov-aerospike) [inline pkg/service/backup_backend_local.go:114]: In the  RemoveFiles branch, when we have only one backup maximum, probably should filter it too, to be consistent.

File: pkg/service/backup_backend_s3.go [hunk 2]
```diff
@@ -88,6 +95,21 @@ func (s *BackupBackendS3) FullBackupList() ([]model.BackupDetails, error) {
 	return result, err
 }
 
+// filterFolders returns backup folders that were created in the specified time range.
+func filterFolders(folders []types.CommonPrefix, from, to int64) ([]types.CommonPrefix, error) {
```
Comment (korotkov-aerospike) [inline pkg/service/backup_backend_s3.go:99]: We have method s.GetTime, that fetches time from S3 (and cache it)
Here you fetch time based on folder name.
I think, we should use the same approach in both places.

Comment (reugn) [reply pkg/service/backup_backend_s3.go:99]: I don't want to make this extra API call when pre-filtering.

File: internal/server/backup_handlers.go [hunk 3]
```diff
@@ -13,12 +15,22 @@ import (
 // @Tags     Backup
 // @Produce  json
 // @Param    name query string false "Backup routine name"
+// @Param    from query int false "Lower bound timestamp filter"
```
Comment (korotkov-aerospike) [inline internal/server/backup_handlers.go:18]: Should've mark it as int64, but I could not find way to do it

Comment (reugn) [reply internal/server/backup_handlers.go:18]: There is no int64/long type in Swagger. int64 will result in the integer type, too. Do you encounter type incompatibility issues generating the client?

Comment (korotkov-aerospike) [reply internal/server/backup_handlers.go]: I found it!

File: pkg/service/backup_backend_local.go [hunk 4]
```diff
@@ -100,6 +100,10 @@ func (local *BackupBackendLocal) FullBackupList() ([]model.BackupDetails, error)
 		if local.fullBackupInProgress.Load() {
 			return []model.BackupDetails{}, nil
 		}
+		// check request time boundaries
+		if lastRun.UnixMilli() < from || lastRun.UnixMilli() >= to {
```
Comment (korotkov-aerospike) [inline pkg/service/backup_backend_local.go:104]: same for backupbackendS3

**Review by korotkov-aerospike — APPROVED** [review]

**Pull request merged**